### PR TITLE
receiver/azureblobreceiver First PR

### LINKED ISSUE #8834: New component: Azure Blob Receiver
## The purpose and use-cases of the new component
This receiver reads logs and traces from [Azure Blob Storage](https://azure.microsoft.com/services/storage/blobs/).

Correlates with #8823 

## Example configuration for the component
The following settings are required:

- `connection_string` (no default): Azure Blob Storage connection key, which can be found in the Azure Blob Storage resource on the Azure Portal.
- `event_hub_end_point` (no default): Azure Event Hub endpoint triggering on the `Blob Create` event 

The following settings can be optionally configured:

- `logs_container_name` (default = "logs"): Name of the blob container with the logs
- `traces_container_name` (default = "traces"): Name of the blob container with the traces

Example:

```yaml
receivers:
  azureblob:
    connection_string: DefaultEndpointsProtocol=https;AccountName=accountName;AccountKey=+idLkHYcL0MUWIKYHm2j4Q==;EndpointSuffix=core.windows.net
    event_hub_end_point: Endpoint=sb://oteldata.servicebus.windows.net/;SharedAccessKeyName=otelhubbpollicy;SharedAccessKey=mPJVubIK5dJ6mLfZo1ucsdkLysLSQ6N7kddvsIcmoEs=;EntityPath=otellhub    
```

The receiver subscribes [on the events](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-event-overview) published by Azure Blob Storage and handled by Azure Event Hub. When it receives `Blob Create` event, it reads the logs or traces from a corresponding blob and deletes it after processing.

## Telemetry data types supported
Logs, Traces

## Sponsor
Sponsor is needed

### PR DESCRIPTION
**Pull request by eedorenko**
Description:
azureblobreceiver reads logs and traces from Azure Blob Storage.

Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8834

Documentation:
Readme describing receiver functionality and configuration.

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
@pmm-sumo please review

**Comment by codeboten** [comment]
pmm-sumo is no longer involved in the project. @mx-psi as the next sponsor on the list, please review this component.

File: receiver/azureblobreceiver/README.md
Comment (mx-psi) [inline receiver/azureblobreceiver/README.md:10]: We usually use just `endpoint` for this. If there could be several endpoints then I would use:

```
event_hub:
  endpoint:
```

File: receiver/azureblobreceiver/README.md
Comment (mx-psi) [inline receiver/azureblobreceiver/README.md:15]: Do you think more options specific for logs or traces could be added in the future? If so, I would create sections for these:

```
logs:
  container_name: logs
traces:
  container_name: traces
```

File: receiver/azureblobreceiver/factory.go
Comment (mx-psi) [inline receiver/azureblobreceiver/factory.go:93]: I believe the way we usually do this is with this module: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/internal/sharedcomponent/sharedcomponent.go You can take a look at how the OTLP receiver does this here: https://github.com/open-telemetry/opentelemetry-collector/blob/27fdc2e2ded2b0900d165edebbf4c26ffc83c2d0/receiver/otlpreceiver/factory.go#L90-L92

File: receiver/azureblobreceiver/go.mod
Comment (mx-psi) [inline receiver/azureblobreceiver/go.mod:5]: We need to move this to the latest released version, `v0.56.0`

**Review by mx-psi — CHANGES_REQUESTED** [review]

File: go.mod
Comment (mx-psi) [inline go.mod:828]: I don't understand the relation of this change with the rest of the PR

File: receiver/azureblobreceiver/factory.go
Comment (mx-psi) [inline receiver/azureblobreceiver/factory.go:112]: You should handle this error

File: receiver/azureblobreceiver/receiver.go
Comment (mx-psi) [inline receiver/azureblobreceiver/receiver.go:35]: Does this need to be public?

File: receiver/azureblobreceiver/README.md
Comment (mx-psi) [inline receiver/azureblobreceiver/README.md:18]: For consistency with existing components this should be `endpoint`, not `end_point`

**Review by mx-psi — COMMENTED** [review]
Looking good, just a few more comments

File: receiver/azureblobreceiver/factory.go
Comment (eedorenko) [reply receiver/azureblobreceiver/factory.go:112]: It is handled [here](https://github.com/eedorenko/opentelemetry-collector-contrib/blob/0744f67d2c2e9c0470a72faad98580f31a283197/receiver/azureblobreceiver/factory.go#L116)

File: go.mod
Comment (eedorenko) [reply go.mod]: There is no relation. Accidentally.

File: receiver/azureblobreceiver/factory.go
Comment (mx-psi) [reply receiver/azureblobreceiver/factory.go]: Oh, okay, thanks. I got confused :)

File: receiver/azureblobreceiver/config.go
Comment (mx-psi) [inline receiver/azureblobreceiver/config.go:21]: Can you open a follow up PR to document the fields directly in the Go struct? This is not urgent, but will be useful in the future for automatic configuration schema generation

**Review by mx-psi — APPROVED** [review]
LGTM

**Comment by mx-psi** [comment]
You need to add this component to `.github/CODEOWNERS`. You can list yourself and also me as the sponsor.

```
FAIL: "receiver/azureblobreceiver" not included in CODEOWNERS
```

File: .github/CODEOWNERS
Comment (mx-psi) [inline .github/CODEOWNERS:139]: Przemek is no longer involved in the project:

```suggestion
receiver/azureblobreceiver/                          @open-telemetry/collector-contrib-approvers @eedorenko @mx-psi
```

**Review by mx-psi — APPROVED** [review]

**Comment by mx-psi** [comment]
Looks good, will merge once all tests pass (probably tomorrow morning first thing if nobody else has done it :))

**Pull request merged**
